src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessor.java

```diff
@@ -77,29 +80,49 @@ public void continueProcessing(UUID envelopeId, BlobClient blob) {
             blob.getContainerName()
         );
 
-        handle(blob, () -> envelopeId);
+        handle(
+            blob,
+            () -> envelopeId,
+            new Condition(
+                () -> envelopeService.findEnvelope(envelopeId).filter(e -> e.status == Status.CREATED).isPresent(),
+                "Envelope is not in the CREATED status"
+            )
+        );
     }
 
-    private void handle(BlobClient blobClient, Supplier<UUID> envelopeIdSupplier) {
+    private void handle(
+        BlobClient blobClient,
+        Supplier<UUID> envelopeIdSupplier,
+        Condition postLeaseCondition
+    ) {
         leaseAcquirer
             .acquireFor(blobClient)
             .ifPresentOrElse(
                 leaseClient -> {
-                    UUID id = envelopeIdSupplier.get();
-                    try {
-                        byte[] rawBlob = downloadBlob(blobClient);
-
-                        var verificationResult = blobVerifier.verifyZip(blobClient.getBlobName(), rawBlob);
-
-                        if (verificationResult.isOk) {
-                            dispatch(blobClient, id, rawBlob);
-                        } else {
-                            reject(blobClient, id, verificationResult.error);
+                    if (postLeaseCondition.isMet()) {
+                        UUID id = envelopeIdSupplier.get();
+                        try {
+                            byte[] rawBlob = downloadBlob(blobClient);
+
+                            var verificationResult = blobVerifier.verifyZip(blobClient.getBlobName(), rawBlob);
+
+                            if (verificationResult.isOk) {
+                                dispatch(blobClient, id, rawBlob);
+                            } else {
+                                reject(blobClient, id, verificationResult.error);
+                            }
+                        } catch (Exception exception) {
+                            handleError(id, blobClient, exception);
+                        } finally {
+                            tryToReleaseLease(leaseClient, blobClient.getBlobName(), blobClient.getContainerName());
                         }
-                    } catch (Exception exception) {
-                        handleError(id, blobClient, exception);
-                    } finally {
-                        tryToReleaseLease(leaseClient, blobClient.getBlobName(), blobClient.getContainerName());
+                    } else {
+                        logger.info(
```

Comment:
Do we check somewhere if the envelope is marked as deleted?

Reply:
no